docs/pages/reference.mdx

```diff
@@ -35,3 +35,59 @@
   ```
   pnpm mud set-version --commit <hash> && pnpm install
   ```
+
+## Debugging
+
+### Onchain components
+
+#### Foundry debugging
+
+The MUD onchain components typically use the [Foundry development stack](https://book.getfoundry.sh/), meaning any of Foundry's debugging tools are available.
+Use this prodecure to view the output from [`anvil`](https://book.getfoundry.sh/reference/anvil/):
+
+1. Add `console.log` to your systems, edit their Solidity files.
+
+   - Import console.log.
+
+     ```solidity
+     import { console } from "forge-std/console.sol";
+     ```
+
+   - Add `console.log` statements inside your Solidity functions.
+
+     ```solidity
+     console.log("newValue:", newValue);
+     ```
+
+1. Run `anvil` manually.
+
+   ```sh
+   anvil -b 1 --base-fee 0 | uniq | tee anvil.log
+   ```
+
+   Note that due to a bug, there might be multiple copies of the same `console.log` message.
+   Piping through [`uniq`](https://man7.org/linux/man-pages/man1/uniq.1.html) helps reduce this issue.
+
+   You do not have to use `tee anvil.log`, but `anvil` logs so much that it is often useful to be able to look for information in the log file after it has scrolled off the screen.
+
+1. Edit `packages/contracts/package.json` to change the `scripts.dev` definition.
+
+   ```
+   "dev": "pnpm mud dev-contracts --tsgenOutput ../client/src/mud --rpc http://127.0.0.1:8545",
```

Comment:
Why? This is for development purposes, don't we want to update the client fields when the data schema is update?

Reply:
it's not needed anymore because we infer RECS components + types from the config: https://github.com/latticexyz/mud/blob/main/.changeset/pretty-hotels-drop.md